File: archive/extract_docx.js
/**
 * extract_docx.js — One-off utility to extract plain text from the trading playbook .docx
 *
 * Node.js alternative to extract_docx.py, using PowerShell's built-in zip extraction.
 * Not part of the running system.
 *
 * Usage:
 *   node archive/extract_docx.js
 */
const fs   = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const docxPath = path.join(__dirname, 'Complete_Trading_System_v3.docx');
const tmpDir   = path.join(__dirname, '..', 'docx_extract_tmp');

if (fs.existsSync(tmpDir)) fs.rmSync(tmpDir, { recursive: true, force: true });

execSync(`powershell -Command "Expand-Archive -Path '${docxPath}' -DestinationPath '${tmpDir}' -Force"`);

let xml = fs.readFileSync(path.join(tmpDir, 'word', 'document.xml'), 'utf-8');
xml = xml.replace(/<\/w:p>/g, '\n').replace(/<\/w:tr>/g, '\n');
let text = xml.replace(/<[^>]+>/g, '');
text = text.replace(/\n\s*\n+/g, '\n\n');

const outPath = path.join(__dirname, '..', 'playbook_extracted.txt');
fs.writeFileSync(outPath, text, 'utf-8');
console.log(`Extracted ${text.length} characters → playbook_extracted.txt`);

fs.rmSync(tmpDir, { recursive: true, force: true });

File: archive/extract_docx.py
"""
extract_docx.py — One-off utility to extract plain text from the trading playbook .docx

Used during development to extract the playbook content for prompt engineering.
Not part of the running system.

Usage:
    python archive/extract_docx.py
    # Outputs: playbook_extracted.txt in the project root
"""
import zipfile
import re
import os

docx_path = os.path.join(os.path.dirname(__file__), 'Complete_Trading_System_v3.docx')

with zipfile.ZipFile(docx_path) as z:
    xml = z.read('word/document.xml').decode('utf-8')

xml = xml.replace('</w:p>', '\n')
xml = xml.replace('</w:tr>', '\n')
text = re.sub(r'<[^>]+>', '', xml)
text = re.sub(r'\n\s*\n+', '\n\n', text)

out_path = os.path.join(os.path.dirname(__file__), '..', 'playbook_extracted.txt')
with open(out_path, 'w', encoding='utf-8') as f:
    f.write(text)

print(f"Extracted {len(text)} characters → playbook_extracted.txt")
